### ISSUE TITLE
Alexa does not realize her own output ?

### ISSUE THREAD

**Issue by antanisblinda** [alexa/alexa-avs-sample-app#371]
Hi. I got the latest version installed on an RPI2 with Raspbian.
Alexa is absolutely amazing. There is something however that i cannot quite understand. 
If i ask Alexa questions about stuff (weather, time, random google questions, ...) it works perfectly. However when I ask other questions (flash briefing, what movies are in theater, ...) alexa would say : "here the news/movies:", pause for a couple of seconds and don't say anything. 
It does look like alexa however thinks that it did say something. For example, at the end it would say "do you want to know about any other movie" even though - in fact - she did not talk about any. 

Also, if I look at the iOS app, the content of the query is there: for example, movies in theaters.

any idea what is going on??

thank you!
this is a fabulous piece of software.

**Comment by alexsalnikov** [comment]
Alexa work in the way that it's recording for some number of seconds microphone input and sends wav file to AVS server where it's try to recognize your speech, then depends on recognition results it's send back wav file with answer, you can access AVS server front end page http://alexa.amazon.com/spa/index.html
and see there what Alexa AVS received as audio input and its answeer, this could help to pinpoint your problem

**Comment by antanisblinda** [comment]
thanks for the reply.
Alexa does understand what I ask ("new movies", or "news flash", ...) but it does not give the audio output of it. If i check the URL you mention or the IOS companion app, i see all my questions there with the right output. So she basically "thinks" she gave me the answer but she never did through the speaker.
Another example is radio: if i use IOS to have alexa play a radio, I do not receive any error, the ios app would show that alexa is playing, but no sound will come out the speaker. More freakingly, if i say "Alexa skip this song", alexa would answer something like "I cannot do that with radio". :)
it drives me nuts!
Other queries (wheather, traffic, random google questions, ...) work; it seems somehow that when alexa needs to rely on something external (imdb for movies, npr for news, ...) that does not work.

any log I can pull to help troubleshoot this?
thanks!

**Comment by alexsalnikov** [comment]
So you get audio response for simple questions and not to ones that will require external skill set connection ?
i would guess or Amazon restricted this sample or you need to enable it somewhere.
sorry i don't know the answer.

**Comment by antanisblinda** [comment]
that's right. the skills (e.g. npr, economist news, ...) are enabled but nothing is being played by alexa. She does think however, as i noted above, that the content is being played. The app shows the same as well.
thanks

**Comment by keleric** [comment]
Alexa's speech is played in a different way than music and list contents. That's why some things work and others don't. It sounds as though VLC isn't working correctly on your machine. Did you use the automated_install.sh script?

**Comment by antanisblinda** [comment]
why are you saying that VLC is not working? Sorry, but I'm not following.
VLC is indeed working. It appears that it is something related to the alexa app not the overall raspbian environment. everything else works fine. thank you.

**Comment by keleric** [comment]
There are 2 media players. VLC, and JLayer. The things that are working for you is everything that uses JLayer. The things that aren't working for you use VLC. Thus, the problem is somehow related to that. VLC may be working, but something else related to it isn't. Maybe the volume on it is muted?

**Comment by antanisblinda** [comment]
Keleric - thank you for the explanation. I see now.
So, I tried to launch vlc from the CLI. At first some output came on screen right before the opening of the app, saying that pulseaudio was not available. Nonetheless VLC opened, and  played a mp3 through my jabra usb. 
like an idiot :), i looked into preferences, audio output and saw that setting was to "automatic", I changed through "also output". It worked 2 other times, closing and re-opening the app.
Now - regardless of what setting i choose, vlc does not play audio anymore. I can't hear anything coming from the jabra.
Any idea on how to reset everything and start again? I apt-get remove --purge vlc and reinstalled but no changes. there is no .vlc folder in my /home/.

thanks

**Comment by antanisblinda** [comment]
not sure if this can be of any help, but here more info.
the above was tried by running vlc as user "pi". vlc refuses to run as "root" (I tried).
when launching alexa services, i execute as normal user (pi) npm start, and it works (listening on port 3000).
then i execute sudo mvn exec:exec, do the token thing, and at that point alexa works the way described in the 1st post ("what's the weather" works, while "give me the news" does not). I'm wondering if it is because mvn tries to use vlc as "root" (given the sudo) and vlc refuses to run.

the reason i do sudo mvn is that if use "pi", the browser is launched, i can login into amazon, but then the little java app (the one with the "listen" button) would not show the token, and the npm terminal would say
error:  { [Error: Not Found] status: 404 }

Does it make any sense to you? should I try to fix the sudo error first? thank you!

**Comment by keleric** [comment]
error: { [Error: Not Found] status: 404 } is a known issue - it doesn't actually affect authorization (and it was fixed on the most recent release). Basically, the web browser is trying to load /favicon.ico, and that 404s.

I suggest downloading the latest version of the sample app, running automated_install.sh (don't use sudo!) and seeing if that fixes your problems.

**Comment by SoNameMuchWow** [comment]
May be a bit late, but I encountered the same issue and it seems to be working now.
As @antanisblinda mentioned, the vlc refuses to run as root so you have to start execute the mvn execute:execute normally (without sudo!). You will then probably get an error about the config.json.
To fix this, simply "sudo chmod 777 config.json" in the javaclient directory as proposed in #191 
Hope this helps everyone struggling with the news / radio stuff.
